x-pack/plugins/security_solution/server/lib/detection_engine/schemas/rule_converters.ts

```diff
@@ -276,6 +279,9 @@ export const commonParamsCamelToSnake = (params: BaseRuleParams) => {
     version: params.version,
     exceptions_list: params.exceptionsList,
     immutable: params.immutable,
+    related_integrations: params.relatedIntegrations ?? [],
+    required_fields: params.requiredFields ?? [],
+    setup: params.setup ?? '',
```

Comment:
I chatted with @spong on Slack about how these fields are currently not included in the create/update APIs and future plans to include them in those APIs. I think that's fine for now. When we add the fields to those APIs, I would recommend coming back to this section and replacing the default values at conversion time with a rule migration to set default values for all rules and making the fields required in the [internal rule schema](https://github.com/elastic/kibana/pull/132409/files#diff-6558622a459138782933b70f1fd34cfba0c55eaa2e75290b8bbf63a6b27efc65R111-R113) - right now these fields are kind of halfway between defaultable and optional - optional internally, but defaultable in the HTTP API. 

I don't have a strong opinion on optional vs defaultable in general, but it's easier to reason about if each field is either optional or defaultable throughout the internal and HTTP schemas. Since we're starting with the fields being required in the response, changing them to optional would be a breaking change, so long term it seems that defaultable + rule migration is the simplest way forward.

Reply:
++, I would prefer if these move to defaultable and behave like the other 'advanced' fields like `author` and `license` (defaulting to empty array and emptystring respectively).

Reply:
++ agree with you @marshallmain! We intentionally decided to make it that way to avoid writing a migration for now. When we add support for these fields to API requests, we will rework it as you suggested. Fewer `null` and `undefined` values here and there - fewer bugs 🙂